### DISCUSSION TITLE
Training documenation

### DISCUSSION THREAD

**Discussion by Technerder** [dscripka/openWakeWord#11] [General]
Are there any plans to release documentation relating to the training of custom models?

**Comment by dscripka** [comment]
Yes, I am planning on releasing more information on how to train new models soon (hopefully in the next few weeks).

Because there is so much data involved (easily 100's of GBs), I won't be able to share everything in exactly the same way that it is being used to produce the models. But I plan to share example notebooks demonstrating the data generation, preparation, and model training with a small set of example data to at least demonstrate the concept.

In the meantime, I'm happy to answer any questions you have about training the models. You can also see a little bit of detail [here](https://github.com/dscripka/openWakeWord/blob/main/docs/synthetic_data_generation.md) on the text-to-speech models used for training data generation, and some more detail about the data preparation in the model documentation [here](https://github.com/dscripka/openWakeWord/tree/main/docs/models).

**Comment by dscripka** [comment]
@Technerder , sorry for the delay, but I'm now close to releasing more information on training new models. I'm still finishing the example notebooks and documentation, but in the meantime I've released the synthetic data generation code in a separate repository here: https://github.com/dscripka/synthetic_speech_dataset_generation

Feel free to experiment with that code, and if you run into any problems create an issue on that repo and I'll try to help resolve it.

**Comment by dscripka** [comment]
@Technerder there is now an example notebook in the repo that demonstrates training custom models: https://github.com/dscripka/openWakeWord/blob/main/notebooks/training_models.ipynb

Feel free to post questions/comments on the custom model training process in this thread.

**Reply by Technerder** [reply]
Awesome! Thank you so much! I'll hopefully be able to try out the notebook in a few days when I have time but at a quick glance it appears to do exactly what I was hoping!

**Reply by jerblack** [reply]
Is all the iron python/jupyter stuff a hard dependency, or can this all be done with pure Python? I'm looking to try running the training locally so I don't have Google shutting down the training in the middle of the session again. Is there already a .py version of this process you've created, or is that something I need to make on my own if I want to run this locally?

**Reply by dscripka** [reply]
@jerblack, no, the jupyter/colab functionality in the notebooks is not a requirement, and you can certainly modify the code to run entirely locally.

[This notebook](https://github.com/dscripka/openWakeWord/blob/main/notebooks/automatic_model_training.ipynb) is a good place to start, as you can see the individual steps (mostly data preparation), and then calling the `train.py` file which does the actual data generation and model training. If you wanted to customize the process for your system, I would recommend reviewing `train.py`.
